src/pymmcore_widgets/hcs/_calibration_widget/_calibration_widget.py

```diff
@@ -0,0 +1,403 @@
+        self._reset_calibration()
+
+        self._calibration_data = value
+
+        self._plate = value.plate if value is not None else None
+
+        # set calibration mode
+        calibration_mode = (
+            self._get_calibration_modes(self._plate.circular_wells)
+            if self._plate is not None
+            else None
+        )
+
+        self._calibration_mode.setValue(calibration_mode)
+
+        # update calibration tables
+        pos_a1 = value.calibration_positions_a1 if value is not None else None
+        pos_an = value.calibration_positions_an if value is not None else None
+        self._update_tables(self._plate, pos_a1, pos_an)
+
+        # update test calibration
+        self._test_calibration.setValue(plate=self._plate, well=None)
+
+        if value is not None and value.calibrated:
+            self._set_calibration_label(True)
+            self._test_calibration._test_button.setEnabled(True)
+
+    def isCalibrated(self) -> bool:
+        """Return True if the plate is calibrated, False otherwise."""
+        return self._calibration_data is not None and self._calibration_data.calibrated
+
+    # _________________________PRIVATE METHODS________________________ #
+
+    def _update_tables(
+        self,
+        plate: WellPlate | None,
+        pos_a1: list[tuple[float, float]] | None,
+        pos_an: list[tuple[float, float]] | None,
+    ) -> None:
+        """Update the calibration tables."""
+        # set table values
+        self._table_a1.setValue([] if pos_a1 is None else pos_a1)
+        self._table_an.setValue([] if pos_an is None else pos_an)
+
+        # set table well name
+        a1 = "" if plate is None else " Well A1 "
+        an = "" if plate is None else f" Well A{plate.columns} "
+        self._table_a1.setLabelText(a1)
+        self._table_an.setLabelText(an)
+
+        # hide an table if plate has only one column
+        self._table_a1.show() if plate is not None else self._table_a1.hide()
+        (
+            self._table_an.show()
+            if plate is not None and plate.columns > 1
+            else self._table_an.hide()
+        )
+
+    def _reset_calibration(self) -> None:
+        """Reset to not calibrated state."""
+        self._set_calibration_label(False)
+        self._test_calibration._test_button.setEnabled(False)
+        self._calibration_data = None
+
+    def _set_calibration_label(self, state: bool) -> None:
+        """Set the calibration label."""
+        lbl_icon = MDI6.check_bold if state else MDI6.close_octagon_outline
+        lbl_icon_size = QSize(20, 20) if state else QSize(30, 30)
+        lbl_icon_color = GREEN if state else RED
+        text = "Plate Calibrated!" if state else "Plate Not Calibrated!"
+        self._calibration_label.setValue(
+            pixmap=icon(lbl_icon, color=lbl_icon_color).pixmap(lbl_icon_size),
+            text=text,
+        )
+
+    def _get_calibration_modes(self, circular_wells: bool) -> list[Mode]:
```

Comment:
don't make this a method, declare it once as a `dict[bool,list[Mode]]` and then `modes = yourdict.get(circular)`